rawfile/analytics/ga4/usage.py

```diff
@@ -1,18 +1,38 @@
 from .client import GA4Client
 from .event import OpenEBSEventBuilder
+from .version_set import VersionSet
+from utils.logs import logger
 
 
 class Usage:
-    def __init__(self, api_secret, measurement_id, client_id):
-        self.client = GA4Client(api_secret, measurement_id, client_id)
+    def __init__(self, api_secret, measurement_id, nodeid):
+        self.nodeid = nodeid
+        self.client = GA4Client(api_secret, measurement_id, nodeid)
         self.builder = OpenEBSEventBuilder()
 
-    def install(self):
-        return self.builder.category("install")
+    def refresh(self):
+        self.builder.refresh_version(VersionSet(self.nodeid))
 
-    def volume_provision(self):
-        return self.builder.category("volume_provision")
+    def send_install(self):
+        self.builder.category("install")
+        self.send()
+
+    def send_ping(self):
+        self.builder.category("ping")
+        self.send()
+
+    def volume_provision(self, name, capacity):
+        self.builder.category("volume_provision").volume_name(name).volume_capacity(
+            capacity
+        ).replica_count(1)
```

Comment:
On the localpvs, we're doing [this](https://github.com/openebs/google-analytics-4/blob/develop/usage/const.go#L21). We're doing that because the older library did that as well. It's a string which says 'replica:1'. There's no need to make any change to what you're doing here, just a heads up.

Reply:
Example so it'd be this?
```json
....,
"vol_replica_count": "replica:1"
```

Reply:
No, `vol_replica_count` is supposed to be 1, there are 3 fields which are common in every custom event, `event_category`, `event_value`, `event_action`, these 3 need to be present in every custom event + whatever field we want.

So for volume creation, the data would be like this,

1. `event_action`: `replica:1`
2. `event_category`: `volume_provision`
3. `event_value`: `50` (size in GB without the unit, just the integer, this is because then in UI it automatically adds it up to show total provision in a day)
4. `vol_replica_count`: `1` --> Custom field for replica count
5. `vol_capacity`: `50Gi` --> Custom field for capacity

The first 3 fields are special as they help up formulate charts later.

Reply:
Discussed offline, the `event_*` are not used in the openebs analytics atm